### PR TITLE
Add store currency support to purchase price threshold filter

### PR DESCRIPTION
**Pull request by eduardoumpierre**
Fixes Automattic/woocommerce-payments-server#3195

#### Changes proposed in this Pull Request

- Add support for store currency on the purchase price threshold filter for fraud protection
- Show store currency on the advanced protection currency inputs

#### Testing instructions

**woocommerce-payments-server**
* Checkout to the branch https://github.com/Automattic/woocommerce-payments-server/pull/3198
* Apply the diff from the PR above

**woocommerce-payments**
* Unit tests should pass
* Update the version on `./woocommerce-payments.php` to be at least `5.7.0`
* Enable the fraud protection feature flag
  * If this is the first time, run `npm run wp option add wcpay_fraud_protection_settings_active 1`
  * If you already have it set, run `npm run wp option update wcpay_fraud_protection_settings_active 1`
* Build the client `npm run start`
* Listen to Stripe webhooks `npm run listen` – or equivalent
* Go to **WooCommerce > Settings**
  * Scroll to **Currency options > Currency**
  * Select a currency as "South African rand (R)"
  * Save the settings
* Click on the **Multi-currency** tab
  * Add **Euro** as a new currency
  * Check the "Add a currency switcher to the Storefront theme on breadcrumb section." box.
  * Save the changes
* Click on the **Payments** tab
  * Go to the "Fraud protection" section
  * Check the "Standard" box and save the settings
  * Check the "Advanced" box and click on "Configure"
  * Scroll down to the "Purchase Price Threshold" section
  * Check the "Screen transactions for abnormal purchase prices" toggle
  * The currency symbol should be "R"
  * The "Maximum purchase price" input should contain the equivalent of US$ 1000 (Around R 17724)
  * Type "1000" on the "Maximum purchase price" input
  * Save the changes
* Go to your shop – add products until the total card price is greater than R 1000.
  * Perform the checkout – it should succeed and the order should be put on hold.
* Go to your shop – change the currency to Euro and add products until the total card price is greater than € 60 (check today's currency exchange) – it should be greater than R 1000.
  * Perform the checkout – it should succeed and the order should be put on hold.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5926` or branch name `add/3195-purchase-price-threshold-store-currency` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/3195-purchase-price-threshold-store-currency) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 7e801de43ddf6a6f6e04a70a4740435f6d221d41
- Build time: 2023-03-31 15:53:08 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/fraud-protection/advanced-settings/cards/test/order-items-threshold.test.js [hunk 1]
```diff
@@ -12,6 +12,25 @@ import OrderItemsThresholdRuleCard, {
 } from '../order-items-threshold';
 
 describe( 'Order items threshold card', () => {
+	beforeEach( () => {
```
Comment (tpaksu) [inline client/settings/fraud-protection/advanced-settings/cards/test/order-items-threshold.test.js]: We won't need this.

File: includes/fraud-prevention/class-fraud-risk-tools.php [hunk 2]
```diff
@@ -170,7 +170,7 @@ public static function get_standard_protection_settings() {
 				Check::check(
 					'order_total',
 					Check::OPERATOR_GT,
-					1000 * 100
+					implode( '|', [ 1000 * 100, 'usd' ] )
```
Comment (tpaksu) [inline includes/fraud-prevention/class-fraud-risk-tools.php]: When I activated the high protection level, and went to the advanced config, I saw this: 
image

Probably we should convert 1000$ into the store currency and write it here?

File: includes/fraud-prevention/class-fraud-risk-tools.php [hunk 3]
```diff
@@ -202,7 +202,7 @@ public static function get_high_protection_settings() {
...
```
Comment (tpaksu) [inline includes/fraud-prevention/class-fraud-risk-tools.php]: Same as the comment I wrote for the standard protection defaults. (Probably that comment should be here and this on the other one 😁 )

**Review by tpaksu — CHANGES_REQUESTED** [review]
Some things to fix, and we'll be good to ship both. Thanks for working on this and bringing it this far!

File: client/settings/fraud-protection/advanced-settings/cards/test/order-items-threshold.test.js (hunk 1 above)
Comment (eduardoumpierre) [reply client/settings/fraud-protection/advanced-settings/cards/test/order-items-threshold.test.js]: Removed on [bfbc55e](https://github.com/Automattic/woocommerce-payments/pull/5926/commits/bfbc55e55cf7ad6f285e9b6653cfc92037dc4140).

File: includes/fraud-prevention/class-fraud-risk-tools.php (hunk 2 above)
Comment (eduardoumpierre) [reply includes/fraud-prevention/class-fraud-risk-tools.php]: That's a good idea! I started to convert the value on the rulesets default value. It was addressed on [2deffc6](https://github.com/Automattic/woocommerce-payments/pull/5926/commits/2deffc6786e11e09449955197b698a19e4c4c6d7).

File: includes/fraud-prevention/class-fraud-risk-tools.php (hunk 3 above)
Comment (eduardoumpierre) [reply includes/fraud-prevention/class-fraud-risk-tools.php]: Thanks for your comment. It was addressed on [2deffc6](https://github.com/Automattic/woocommerce-payments/pull/5926/commits/2deffc6786e11e09449955197b698a19e4c4c6d7).

**Review by tpaksu — APPROVED** [review]
Now all test good! Thanks for fixing the issues on the PR and finishing this! LGTM.

**Pull request merged**